CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -180,26 +172,23 @@ class CameraFragment : Fragment() {
 
             // Load thumbnail into circular button using Glide
             Glide.with(thumbnail)
-                    .load(file)
-                    .apply(RequestOptions.circleCropTransform())
-                    .into(thumbnail)
+                .load(file)
+                .apply(RequestOptions.circleCropTransform())
+                .into(thumbnail)
         }
     }
 
     /** Define callback that will be triggered after a photo has been taken and saved to disk */
-    private val imageSavedListener = object : ImageCapture.OnImageSavedListener {
-        override fun onError(
-                error: ImageCapture.ImageCaptureError, message: String, exc: Throwable?) {
+    private val imageSavedListener = object : ImageCapture.OnImageSavedCallback {
+        override fun onError(imageCaptureError: Int, message: String, cause: Throwable?) {
             Log.e(TAG, "Photo capture failed: $message")
```

Comment:
I would also add stack information to the logs: `Log.e(TAG, "Photo capture failed: $message", cause)`

Reply:
The `cause` had been added.